H5_sampler/sample_signal.py

```diff
@@ -0,0 +1,64 @@
+from Sampler import BlackBox, MaxMultiSampler
+from optparse import OptionParser
+import os
+from os.path import join, dirname, exists
+import numpy as np
+import json
+
+parser = OptionParser()
+parser.add_option("-s", "--seed", type=int, default=123, help="Random seed")
+parser.add_option("-i", "--input", type=str,
+                  default="/eos/cms/store/group/phys_b2g/CASE/h5_files/UL/",
+                  help="Input base directory.")
+parser.add_option("-o", "--output", type=str, default="./mixed_years/",
+                  help="Output directory.")
+parser.add_option("-d", "--subdir", type=str, default="new/",
+                  help=("Subdirectory after the year-specific folder "
+                        "(<input base dir>/<year dir>/<subdir>/<actual h5 input samples>)."
+                        "Provide empty string if subdir should be omitted."))
+parser.add_option("-c", "--config", type=str,
+                  default=join(dirname(__file__), "sample_signal_config.json"),
+                  help="JSON config file containing zears and relative lumies.")
+options, args = parser.parse_args()
+
+
+base_dir = options.input
+out_dir = options.output
+subdir = options.subdir
+with open(options.config) as json_file:
+    config = json.load(json_file)
+    years = config["years"]
+    lumies = config["lumies"]
+np.random.seed(options.seed)
+
+assert len(years) > 1
+for year in years:
+    assert year in lumies.keys()
+
+if not exists(out_dir):
+    os.makedirs(out_dir)
```

Comment:
Since Python 3.2:
```suggestion
os.makedirs(out_dir, exist_ok=True)
```
(but are we even using Python 3?)

Reply:
I also expected this one. I'm running the script in python2 because CMS.